libs/filter_lib/Cargo.toml

```diff
@@ -11,7 +11,7 @@ crate-type = ["dylib"]
 path = "src/lib.rs"
 
 [dependencies]
-petgraph = "0.5.1"
+petgraph = { path = "../petgraph" }
```

Comment:
I would recommend adding petgraph as a git submodule and track changes there. Makes maintenance a lot easier.